exporter/trace/trace.go

```diff
@@ -41,42 +41,57 @@ type traceExporter struct {
 	client *traceclient.Client
 }
 
-const defaultBufferedByteLimit = 8 * 1024 * 1024
 const defaultBundleDelayThreshold = 2 * time.Second
 const defaultBundleCountThreshold = 50
-const bundleByteThresholdMultiplier = 300
-const bundleByteLimitMultiplier = 1000
+const defaultBundleByteThreshold = 15000
+const defaultBundleByteLimit = 0
+const defaultBufferedByteLimit = 8 * 1024 * 1024
 
 func newTraceExporter(o *options) (*traceExporter, error) {
 	client, err := traceclient.NewClient(o.Context, o.TraceClientOptions...)
 	if err != nil {
 		return nil, fmt.Errorf("stackdriver: couldn't initiate trace client: %v", err)
 	}
 	e := &traceExporter{
-		projectID: o.ProjectID,
-		client:    client,
-		o:         o,
+		projectID:      o.ProjectID,
+		client:         client,
+		o:              o,
+		overflowLogger: overflowLogger{delayDur: 5 * time.Second},
 	}
 	b := bundler.NewBundler((*tracepb.Span)(nil), func(bundle interface{}) {
 		e.uploadFn(context.Background(), bundle.([]*tracepb.Span))
 	})
+
 	if o.BundleDelayThreshold > 0 {
 		b.DelayThreshold = o.BundleDelayThreshold
 	} else {
 		b.DelayThreshold = defaultBundleDelayThreshold
 	}
+
 	if o.BundleCountThreshold > 0 {
 		b.BundleCountThreshold = o.BundleCountThreshold
 	} else {
 		b.BundleCountThreshold = defaultBundleCountThreshold
 	}
-	b.BundleByteThreshold = b.BundleCountThreshold * bundleByteThresholdMultiplier
-	b.BundleByteLimit = b.BundleCountThreshold * bundleByteLimitMultiplier
+
+	if o.BundleByteThreshold > 0 {
```

Comment:
Doesn't block this PR, but should we return errors for invalid configuration instead of silently using the default?

Reply:
We should chat about that.  I think for something like it we should do our best to ignore config errors and limp along not making your app worse.

However, this would be an undiscoverable config error, which should be fixed in some fashion...

Reply:
We could just pass these parameters as is to `bundler.NewBundler` and let it handle zero values and other validation.

Reply:
Doesn't look the bundler library does any validation of its own. I think in all their use cases, negative values will behave the same as zero which is arguably an idiomatic way to handle things, e.g. They use `DelayThreshold` in `time.AfterFunc` which ends up hitting this condition in the Go source: https://github.com/golang/go/blob/5f3354d1bf2e6a61e4b9e1e31ee04b99dfe7de35/src/time/sleep.go#L29

Still, I'd be happy to make this return errors instead but would do in a follow up PR

Reply:
Or maybe that makes more sense to be something that is done by the code that reads config, e.g. in the Collector

Reply:
Sorry for not responding sooner. I think the important part here is from a user perspective, there is a correctable notification of failures and values aren't silently converted in a way that's confusing.

Likely that means work in whatever reads the config and passes to this function.